lib/client/known_hosts_migrate.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+)
+
+// knownHostEntry is a parsed entry from a Teleport/OpenSSH known_hosts file,
+// used as part of the migration/pruning process to make Teleport's known_hosts
+// compatible with OpenSSH.
+type knownHostEntry struct {
+	raw     string
+	marker  string
+	hosts   []string
+	pubKey  ssh.PublicKey
+	comment string
+}
+
+// parseKnownHost parses a single line of a known hosts file into a struct.
+func parseKnownHost(raw string) (*knownHostEntry, error) {
+	// Due to the lack of first-class tuples, we'll need to wrap this in a
+	// struct to avoid re-parsing lines constantly. We'll also keep the input
+	// text to preserve formatting for all passed-through entries.
+	marker, hosts, pubKey, comment, _, err := ssh.ParseKnownHosts([]byte(raw))
+	if err != nil {
+		return nil, err
+	}
+
+	return &knownHostEntry{
+		raw,
+		marker,
+		hosts,
+		pubKey,
+		comment,
```

Comment:
Can you name the fields explicitly please?

Reply:
Sure, done.